worldengine/generation.py

```diff
@@ -141,18 +141,53 @@ def harmonize_ocean(ocean, elevation, ocean_level):
 # ----
 
 def sea_depth(world, sea_level):
+
+    # a dynamic programming approach to gather how far the next land is 
+    # from a given coordinate up to a maximum distance of max_radius
+    # result is 0 for land coordinates and -1 for coordinates further than
+    # max_radius away from land
+    # there might be even faster ways but it does the trick
+
+    def next_land_dynamic(ocean, max_radius=5):
+    
+        next_land = numpy.full(ocean.shape, -1)
+
+        # non ocean tiles are zero distance away from next land
+        next_land[numpy.logical_not(ocean)]=0
+
+        height, width = ocean.shape
+
+        for dist in range(max_radius):
+            for y in range(height):
+                for x in range(width):
+                    if next_land[y,x] == -1:
+                        for dx in range(-1, 2):
+                            nx = x + dx
+                            if 0 <= nx < width:
+                                for dy in range(-1, 2):
+                                    ny = y + dy
+                                    if 0 <= ny < height and (dx != 0 or dy != 0):
+                                        if next_land[ny,nx] == dist:
+                                            next_land[y,x] = dist+1
+
+        return next_land
+
+    next_land = next_land_dynamic(world.layers['ocean'].data)
+
     sea_depth = sea_level - world.layers['elevation'].data
     for y in range(world.height):
         for x in range(world.width):
-            if world.tiles_around((x, y), radius=1, predicate=world.is_land):
+            dist_to_next_land = next_land[y,x]
+
+            if dist_to_next_land == 1:
```

Comment:
Maybe we can now use an array containing the different factors and then just do:
```
sea_depth[y, x] *= factor[dist_to_next_land]
```

Reply:
Of course, Sometimes you don't see the simplest things. In my defense it would have been a bit harder to see then, that the code actually behaves the same.